02_knight_and_princess/danil_dyachenko/graph_gen.cpp

```diff
@@ -0,0 +1,122 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  VertexId id;
```

Comment:
Дефолтное значение не помешало бы, в противном случае, есть возможность инициализировать объект с `id` указывающим на какой-то мусор в памяти.

Можно сделать так:
Объявить глобальную константу типа 
```cpp
constexpr int INVALID_ID = -1;
```
```suggestion
  VertexId id = INVALID_ID;
```